Public/Src/Tools/SymbolDaemon/Tool.SymbolDaemonRunner.dsc

```diff
@@ -0,0 +1,296 @@
+        untrackedDirectoryScopes: [
+            ...(tool.untrackedDirectoryScopes || []),
+            ...(args.untrackedDirectoryScopes || []),
+            symbolLogDirectory
+        ],
+    });
+}
+
+function getExecuteArguments(command: string, args: UberArguments, ...additionalCmdArgs: Argument[]): Transformer.ExecuteArguments {
+    const nametag = `symbold-${command}`;
+    const outDir = Context.getNewOutputDirectory(nametag);
+    const selectedTool: Transformer.ToolDefinition = args.tool || tool;
+    Contract.assert(
+        selectedTool !== undefined, 
+        "tool not specified"
+    );
+
+    return <Transformer.ExecuteArguments>{
+        tool: overrideToolArguments(selectedTool, args),
+        workingDirectory: outDir,
+        tags: [
+            symbolTag,
+            nametag,
+            ...(args.tags || []),
+        ],
+        arguments: [
+            Cmd.argument(command),
+            Cmd.option("--moniker ", args.ipcMoniker),
+            Cmd.option("--ipcServerMoniker ", args.ipcServerMoniker),
+            Cmd.option("--name ", args.name),
+            Cmd.option("--service ", args.service),            
+            Cmd.option("--maxConcurrentClients ", args.maxConcurrentClients),
+            Cmd.option("--maxParallelUploads ", args.maxParallelUploads),            
+            Cmd.option("--retentionDays ", args.retentionDays),
+            Cmd.option("--maxConnectRetries ", args.maxConnectRetries),
+            Cmd.option("--connectRetryDelayMillis ", args.connectRetryDelayMillis),
+            Cmd.flag("--enableCloudBuildIntegration", args.enableCloudBuildIntegration),
+            Cmd.flag("--enableTelemetry", args.enableTelemetry),
+            Cmd.flag("--verbose", true),
+            Cmd.option("--logDir ", symbolLogDirectory.path),
+            ...additionalCmdArgs,
+            Cmd.option(
+                "--configFile ",
+                Artifact.input(args.symbolServiceConfigFile)
+            ),
+        ],
+        consoleOutput: outDir.combine(`${nametag}-stdout.txt`),
+        dependencies: [
+            ...(args.dependencies || []),
+        ],
+        environmentVariables: (args.additionalEnvironmentVars || []),
+        unsafe: {
+            passThroughEnvironmentVariables: (args.forwardEnvironmentVars || []),
+        }   
+    };
+}
+
+function executePingAsProcess(serviceStartInfo: ServiceStartResult, args: OperationArguments): Result {
+    const exeArgs = getExecuteArguments("ping", overrideMoniker(args, serviceStartInfo.ipcMoniker)).merge<Transformer.ExecuteArguments>({
+        servicePipDependencies: [ serviceStartInfo.symbolDaemonId ],
+    });
+    const result = Transformer.execute(exeArgs);
+    return <Result>{outputs: result.getRequiredOutputFiles() };
+}
+
+function executeDaemonCommand(serviceStartInfo: ServiceStartResult, command: string, args: UberArguments, overrideIpcArgs?: (a: Transformer.IpcSendArguments) => Transformer.IpcSendArguments ): Result {

```

Comment:
>overrideIpcArgs?: (a: Transformer.IpcSendArguments) => Transformer.IpcSendArguments [](start = 106, length = 83)

since all 2 use cases are using merge...
have you considered just passing 
additionaIpcArgs?: IpcSendArguments and merge in this function?

Reply:
I don't think I can. there are 3 calls -- 2 of them use merge, and the other one uses override

---
In reply to: [313494670](https://github.com/microsoft/BuildXL/pull/727#discussion_r313494670) [](ancestors = 313494670)